src/main.c

```diff
@@ -599,10 +599,11 @@ static void core_secondly()
       check_botnet_pings();
 
       if (!miltime) {           /* At midnight */
-        char s[25];
+        char s[26];
         int j;
 
-        strlcpy(s, ctime(&now), sizeof s);
+        ctime_r(&now, s);
+        s[24] = 0;
```

Comment:
why is it an array of 26 if you cut it off at index 24?

Reply:
because ctime_r will write 26 bytes. if buffer smaller, then bufferoverflow.

see also:

https://manpages.debian.org/bookworm/manpages-dev/ctime_r.3.en.html

char *ctime_r(const time_t *restrict timep,
                    char buf[restrict 26]);

Reply:
truncating it with `s[24] = 0` is to strip the trailing `\n`